SARSA raises an error with GPU: ValueError: Unsupported dtype object

`tests/agents_tests/test_sarsa.py`'s slow gpu tests fail.

```
tests/agents_tests/test_sarsa.py::TestSARSAOnDiscreteABC::test_batch_training_gpu <- tests/agents_tests/basetest_training.py FAILED

=============================================================================================================================== FAILURES ===============================================================================================================================
____________________________________________________________________________________________________________ TestSARSAOnDiscreteABC.test_batch_training_gpu ____________________________________________________________________________________________________________

self = <test_sarsa.TestSARSAOnDiscreteABC testMethod=test_batch_training_gpu>

    @testing.attr.slow
    @testing.attr.gpu
    def test_batch_training_gpu(self):
>       self._test_batch_training(0, steps=100000)

tests/agents_tests/basetest_training.py:170:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/agents_tests/basetest_training.py:146: in _test_batch_training
    eval_env=test_env,
chainerrl/experiments/train_agent_batch.py:215: in train_agent_batch_with_evaluation
    logger=logger)
chainerrl/experiments/train_agent_batch.py:77: in train_agent_batch
    agent.batch_observe_and_train(obss, rs, dones, resets)
chainerrl/agents/dqn.py:512: in batch_observe_and_train
    self.replay_updater.update_if_necessary(self.t)
chainerrl/replay_buffer.py:289: in update_if_necessary
    self.update_func(transitions)
chainerrl/agents/dqn.py:269: in update
    loss = self._compute_loss(exp_batch, errors_out=errors_out)
chainerrl/agents/dqn.py:349: in _compute_loss
    y, t = self._compute_y_and_t(exp_batch)
chainerrl/agents/dqn.py:335: in _compute_y_and_t
    self._compute_target_values(exp_batch),
chainerrl/agents/sarsa.py:35: in _compute_target_values
    for i in range(len(exp_batch['action']))])
../.local/share/virtualenvs/chainerrl_20180827-9TPAa4mI/lib/python3.5/site-packages/cupy/creation/from_data.py:41: in array
    return core.array(obj, dtype, copy, order, subok, ndmin)
cupy/core/core.pyx:1714: in cupy.core.core.array
    ???
cupy/core/core.pyx:1791: in cupy.core.core.array
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   ValueError: Unsupported dtype object
```

This is because `cupy.ndarray` and python int are passed to cupy.array at the same time, which results in ValueError.

--- a/chainerrl/agents/sarsa.py
+++ b/chainerrl/agents/sarsa.py
@@ -6,6 +6,8 @@ from builtins import *  # NOQA
 from future import standard_library
 standard_library.install_aliases()  # NOQA
 
+import chainer
+
 from chainerrl.agents import dqn
 
 
@@ -27,9 +29,11 @@ class SARSA(dqn.DQN):
         else:
             target_next_qout = self.target_model(batch_next_state)
         # Choose an action using the behavior policy
+        next_greedy_actions = chainer.cuda.to_cpu(
+            target_next_qout.greedy_actions.array)
         batch_next_action = self.xp.array([
             self.explorer.select_action(
-                self.t, lambda: target_next_qout.greedy_actions.array[i],
+                self.t, lambda: next_greedy_actions[i],
                 action_value=target_next_qout[i:i + 1],
             )
             for i in range(len(exp_batch['action']))])

